File: 프로그래머스/unrated/176963. 추억 점수/추억 점수.cpp
#include <bits/stdc++.h>

using namespace std;

vector<int> solution(vector<string> name, vector<int> yearning, vector<vector<string>> photo) {
    vector<int> ans;
	map<string, int> name_map;

	for(int i=0;i<name.size();i++){
		name_map.insert({name[i], yearning[i]});	
	}

	for(auto p : photo){
		int sum = 0;
		for(auto t : p){
			if(name_map.find(t) != name_map.end())
				sum+=name_map.find(t)->second;
		}
		ans.push_back(sum);
	}

    return ans;
}

File: 프로그래머스/unrated/176963. 추억 점수/추억 점수.java
import java.util.*;

class Solution {
    public int[] solution(String[] name, int[] yearning, String[][] photo) {
        int[] answer = new int[photo.length];
        HashMap<String, Integer> nameMap = new HashMap<>();

        for(int i=0;i<name.length;i++){
            nameMap.put(name[i], yearning[i]);
        }

        for(int i=0;i<photo.length;i++){
            int sum = 0;
            for(String p : photo[i]){
                if(nameMap.containsKey(p))
                    sum += nameMap.get(p);
            }
            answer[i] = sum;
        }
        return answer;
    }
}